htdocs/core/lib/files.lib.php

```diff
@@ -1556,7 +1556,8 @@ function dol_add_file_process($upload_dir, $allowoverwrite=0, $donotupdatesessio
 				$destfull = $info['dirname'].'/'.dol_sanitizeFileName($info['filename'].'.'.strtolower($info['extension']));
 				$info = pathinfo($destfile);
 				$destfile = dol_sanitizeFileName($info['filename'].'.'.strtolower($info['extension']));
-
+				$destfile = dol_string_nohtmltag($destfile);
```

Comment:
dol_string_nohtmltag is a function to escape html content. We should have no reason to apply it on a file path.
Can you provide example of case you want to fix with this ?

Reply:
Hi @eldy 

If you upload a file with multiple consecutive spaces in the name as "file-example__1.pdf" (for the exemple replace _ by space) we got an error when we try to download  
https://github.com/Dolibarr/dolibarr/blob/9de0e694569d82c2b5110c36917d3ae49c37e887/htdocs/document.php#L77 
In `GETPOST` function
https://github.com/Dolibarr/dolibarr/blob/9de0e694569d82c2b5110c36917d3ae49c37e887/htdocs/core/lib/functions.lib.php#L540

In `dol_string_nohtmltag` function
https://github.com/Dolibarr/dolibarr/blob/9de0e694569d82c2b5110c36917d3ae49c37e887/htdocs/core/lib/functions.lib.php#L5393

Reply:
Hi @eldy , @atm-ph is right.
I want to use this function because it converts all multiple consecutive spaces in only one.
It avoids getting error when we want to download file later.